### ISSUE TITLE
Xperifirm is not launching

### ISSUE THREAD

**Issue by churchner420** [Androxyde/Flashtool#163]
When pressing the XperFirm button in the flashtool menu nothing happens. Bundles therefore have to created manually

**Comment by misaki1301** [comment]
still xperifirm not open, downloaded from xdadevelopers but for some reason doesn't detect the Stock ROM downloaded

**Comment by 0com** [comment]
If you install 0.9.18.6 first and run xperifirm once, you can then install the latest flashtool version in the same folder with the xperifirm finally working.

**Comment by churchner420** [comment]
> 
> 
> If you install 0.9.18.6 first and run xperifirm once, you can then install the latest flashtool version in the same folder with the xperifirm finally working.

Yes thanks that worked for me. But when I am trying to flash my 5 II now it says "no flash script found". How can I fix that?

**Comment by 0com** [comment]
> > If you install 0.9.18.6 first and run xperifirm once, you can then install the latest flashtool version in the same folder with the xperifirm finally working.
> 
> Yes thanks that worked for me. But when I am trying to flash my 5 II now it says "no flash script found". How can I fix that?

That sometimes happens when you manually build the bundle. In that case it always work for me to help the flashtool to find the script by renaming it by the same version as the rom. For example if you use a rom ver 41.1.0-whatever and the script name is 41.fsc, renamed it to 41.1.0-whatever.fsc 

Also if you get constant nonsensical errors, please try with another / shorter cable. Recently I lost one entire day trying everything to flash a rom and at the end the only issue was the usb cable lol. Good luck!

**Comment by lzy0702** [comment]
> If you install 0.9.18.6 first and run xperifirm once, you can then install the latest flashtool version in the same folder with the xperifirm finally working.

Now it shows 

> 01/039/2022 03:39:13 - ERROR - http://www.iagucool.com/xperifirm/version

**Comment by NightMean** [comment]
I've submitted a PR to fix this but I doubt it will get merged. 

What you can do manually is: 
1. Download the latest version from XDA: https://xdaforums.com/t/tool-xperifirm-xperia-firmware-downloader-v5-7-1.2834142/
2. Extract it and place the `XperiFirm-64.exe` or `XperiFirm-x86.exe` into `C:\Users\<YOUR_USERNAME>\.flashTool`. (Replace the YOUR_USERNAME with your actual user on the PC) 
3. Rename the `XperiFirm-64.exe` or `XperiFirm-x86.exe` to **`XperiFirm.exe`** otherwise it will not get recognized
4. After this, the button in Flashtool will work as expected.

**Comment by MartinVonReichenberg** [comment]
This is the official URL, by the way:
https://xperifirmtool.com/
